autogpt_platform/frontend/src/components/cronScheduler.tsx

```diff
@@ -0,0 +1,434 @@
+              </div>
+              {showCustomDays && (
+                <div className="flex flex-wrap gap-2">
+                  {Array.from({ length: 31 }, (_, i) => (
+                    <Button
+                      key={i + 1}
+                      variant={
+                        selectedDays.includes(i + 1) ? "default" : "outline"
+                      }
+                      className="h-10 w-10 p-0"
+                      onClick={() => {
+                        setSelectedDays((prev) =>
+                          prev.includes(i + 1)
+                            ? prev.filter((d) => d !== i + 1)
+                            : [...prev, i + 1],
+                        );
+                      }}
+                    >
+                      {i + 1}
+                    </Button>
+                  ))}
+                </div>
+              )}
+            </div>
+          )}
+          {frequency === "yearly" && (
+            <div className="space-y-4">
+              <Label>Months</Label>
+              <div className="flex gap-2">
+                <Button
+                  variant="outline"
+                  className="h-8 px-2 py-1 text-xs"
+                  onClick={() => {
+                    if (selectedDays.length === months.length) {
+                      setSelectedDays([]);
+                    } else {
+                      setSelectedDays(Array.from({ length: 12 }, (_, i) => i));
+                    }
+                  }}
+                >
+                  {selectedDays.length === months.length
+                    ? "Deselect All"
+                    : "Select All"}
+                </Button>
+              </div>
+              <div className="flex flex-wrap gap-2">
+                {months.map((month, index) => (
+                  <Button
+                    key={index}
+                    variant={
+                      selectedDays.includes(index) ? "default" : "outline"
+                    }
+                    className="px-2 py-1"
+                    onClick={() => {
+                      setSelectedDays((prev) =>
+                        prev.includes(index)
+                          ? prev.filter((m) => m !== index)
+                          : [...prev, index],
+                      );
+                    }}
+                  >
+                    {month.label}
+                  </Button>
+                ))}
+              </div>
+            </div>
+          )}
+
+          {frequency !== "minute" && frequency !== "hour" && (
+            <div className="flex items-center gap-4 space-y-2">
+              <Label className="pt-2">At</Label>
+              <Input
+                type="time"
+                value={selectedTime}
+                onChange={(e) => setSelectedTime(e.target.value)}
+              />
+            </div>
+          )}
+
+          <Separator />
+          {/*
+
+            On the backend, we are using standard cron expressions,
```

Comment:
This is a really good idea, I can accommodate this on my API https://github.com/Significant-Gravitas/AutoGPT/pull/8649, what field do you think you need for this? I can add end_time on the `GET` `PUT` & `POST` request wdyt?